lib/engine/step/g_18_mex/swap_buy_sell.rb

```diff
@@ -30,8 +30,9 @@ def swap_sell(player, corporation, bundle, pool_share)
   # Private method used by other methods in this module
   def bundle_reduced_five_percent(shares)
     # Dup is needed to avoid affecting the actual percentage in the original bundle
-    updated_bundle = Engine::ShareBundle.new(shares.map(&:dup))
-    updated_bundle.shares.first.percent -= 5
+    updated_shares = shares.map(&:dup)
+    updated_shares.first.percent -= 5
+    updated_bundle = Engine::ShareBundle.new(updated_shares)
```

Comment:
Could possibly merge line 35 and 36 to just
Engine::ShareBundle.new(updated_shares)